Source/Core/Duality/Utility/Coroutines/CoroutineManager.cs

```diff
@@ -0,0 +1,115 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Duality.Utility.Coroutines
+{
+	/// <summary>
+	/// The Coroutine Manager
+	/// </summary>
+	public class CoroutineManager
+	{
+		private readonly Queue<Coroutine> pool = new Queue<Coroutine>(64);
+		private readonly Queue<Coroutine> coroutines = new Queue<Coroutine>(256);
+		private readonly Queue<Coroutine> nextCycle = new Queue<Coroutine>(256);
+		private readonly Queue<Coroutine> scheduled = new Queue<Coroutine>();
```

Comment:
Do we really need to have a separate sheduled and a coroutines queue? Isnt it possible to just have 1 coroutine queue?